cpp/tensorrt_llm/thop/groupRmsNormOp.cpp

```diff
@@ -0,0 +1,198 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tensorrt_llm/common/cudaUtils.h"
+#include "tensorrt_llm/common/dataType.h"
+#include "tensorrt_llm/common/opUtils.h"
+#include "tensorrt_llm/kernels/groupRmsNormKernels/groupRmsNormKernels.h"
+#include "tensorrt_llm/runtime/torchUtils.h"
+#include "tensorrt_llm/thop/thUtils.h"
+
+#include <ATen/cuda/CUDAContext.h>
+#include <cstdint>
+#include <cuda_runtime.h>
+#include <stdexcept>
+#include <torch/extension.h>
+#include <vector>
+
+namespace torch_ext
+{
+
+void groupRMSNorm(torch::TensorList const& inputs, torch::TensorList const& outputs, torch::TensorList const& weights,
+    double eps, double weight_bias)
+{
+    TORCH_CHECK(!inputs.empty(), "Input tensor list cannot be empty.");
+    TORCH_CHECK(!outputs.empty(), "Output tensor list cannot be empty.");
+    uint32_t const num_inputs = inputs.size();
+    TORCH_CHECK(num_inputs <= 2, "Only up to 2 inputs are supported.");
+    auto const first_input = inputs[0];
+    TORCH_CHECK(first_input.dim() == 2, "Inputs must be 2D tensors [batch_size, hidden_dim].");
+    TORCH_CHECK(first_input.sizes()[0] > 0, "Batch size must be greater than 0.");
+    TORCH_CHECK(first_input.is_cuda(), "Inputs must be CUDA tensors.");
+    uint32_t const batch_size = first_input.sizes()[0];
+    auto const dtype = first_input.scalar_type();
+    auto const device = first_input.device();
+
+    if (!weights.empty())
+    {
+        TORCH_CHECK(weights.size() == num_inputs, "Weights list size must match inputs list size.");
+    }
+
+    for (size_t i = 0; i < num_inputs; ++i)
+    {
+        TORCH_CHECK(inputs[i].sizes()[0] == batch_size, "Inputs must have the same batch size.");
+        TORCH_CHECK(inputs[i].dim() == 2, "Inputs must be 2D tensors [batch_size, hidden_dim].");
+        TORCH_CHECK(inputs[i].device() == device, "Inputs must be on the same device.");
+        TORCH_CHECK(inputs[i].scalar_type() == dtype, "Inputs must be of the same type.");
+        TORCH_CHECK(outputs[i].dim() == 2, "Outputs must be 2D tensors [batch_size, hidden_dim].");
+        TORCH_CHECK(outputs[i].device() == device, "Outputs must be on the same device.");
+        TORCH_CHECK(outputs[i].scalar_type() == dtype, "Outputs must be of the same type.");
+        TORCH_CHECK(outputs[i].sizes()[0] == batch_size, "Outputs must have the same batch size.");
+        TORCH_CHECK(
+            outputs[i].sizes()[1] == inputs[i].sizes()[1], "Outputs and inputs must have the same last dimension.");
+        TORCH_CHECK(inputs[i].strides()[0] == outputs[i].strides()[0], "Inputs and outputs must have the same stride.");
+        TORCH_CHECK(inputs[i].strides()[1] == 1, "Inputs must be contiguous.");
+        TORCH_CHECK(outputs[i].strides()[1] == 1, "Outputs must be contiguous.");
```

Comment:
```suggestion
        TORCH_CHECK(inputs[i].strides()[1] == 1, "Inputs must be contiguous along the last dim.");
        TORCH_CHECK(outputs[i].strides()[1] == 1, "Outputs must be contiguous along the last dim.");
```
Fix the other place too.